mesop/web/src/services/channel.ts

```diff
@@ -122,8 +123,33 @@ export class Channel {
         const uiResponse = UiResponse.deserializeBinary(array);
         console.debug('Server event: ', uiResponse.toObject());
         switch (uiResponse.getTypeCase()) {
+          case UiResponse.TypeCase.UPDATE_STATE_EVENT: {
+            const states = uiResponse.getUpdateStateEvent()!.getStates()!;
+
+            if (states === undefined) {
+              console.error('Update state event is empty.');
+              return;
+            }
+
+            if (
+              uiResponse.getUpdateStateEvent()!.getUpdateStrategy() ===
+              UpdateStateEvent.UpdateStrategy.UPDATE_STRATEGY_DIFF
+            ) {
+              // `this.states` should be populated at this point since the first update
```

Comment:
This shouldn't happen normally, but it's probably worth checking that this.states and states have the same length. There could catch a weird edge case where someone tries to lazily register a stateclass.

Reply:
Hmm, yeah. How should we handle that case?

Reply:
If the lengths are not the same, then I would just throw an exception, since it's very likely we'd produce the wrong state result anyways.

Reply:
I see. I thought you had some kind of mitigation in mind. Updated.